packages/qdrant-loader/src/qdrant_loader/core/state/state_manager.py

```
@@ -264,6 +264,99 @@ async def mark_document_deleted(
+
+                # Commit state changes immediately
+                await tx.commit()
+                self.logger.info(
+                    f"Marked {len(document_ids_to_delete)} documents as deleted in state DB"
+                )
+            except Exception as e:
+                # Rollback on any DB error
+                try:
+                    await tx.rollback()
+                except Exception as rb_err:
+                    self.logger.error(
```

Ruff (0.15.14)

[warning] Logging `.exception(...)` should be used instead of `.error(..., exc_info=True)`

(G201)